src/util/options-validator.js

```diff
@@ -0,0 +1,39 @@
+const lget = require('lodash.get');
+const BLANK_DEFAULT = {};
+
+class BuildpackValidationError extends Error {
+    constructor(name, callsite, validationErrors) {
+        super();
+
+        const bullet = '\n\t- ';
+        this.name = 'BuildpackValidationError';
+        this.message =
+            `${name}: Invalid configuration object. ` +
+            `${callsite} was called with a configuration object that has the following problems:${bullet}` +
+            validationErrors
+                .map(
+                    ([key, requiredType]) =>
+                        `${key} must be of type ${requiredType}`
+                )
+                .join(bullet);
+        this.validationErrors = validationErrors;
+
+        Error.captureStackTrace(this, this.constructor);
+    }
+}
+
+module.exports = (name, simpleSchema) => (callsite, options) => {
```

Comment:
Why not just use `joi` or JSON schema here? I'd imagine you're going to hit the edges of this pretty quick, and we'll quickly end up in the business of writing an options validator.

Reply:
No need to hold up, but a comment with a TODO would be cool. Easy "good first issue" to throw up later

Reply:
God idea